packages/components/src/unit-control/test/index.tsx

```diff
@@ -323,15 +305,13 @@ describe( 'UnitControl', () => {
 
 			// This is because `isPressEnterToChange` is `true`
 			expect( onChangeSpy ).not.toHaveBeenCalled();
+			expect( onUnitChangeSpy ).not.toHaveBeenCalled();
 
-			// Clicking on the button should cause the `onBlur` callback to fire.
-			const button = screen.getByRole( 'button' );
-			await user.click( button );
-
-			await waitFor( () =>
-				expect( onChangeSpy ).toHaveBeenCalledTimes( 1 )
-			);
+			// Clicking document.body to trigget a blur event on the input.
+			await user.click( document.body );
 
+			// TODO: investigate why `onChange` gets called twice instead of once
+			expect( onChangeSpy ).toHaveBeenCalledTimes( 2 );
```

Comment:
This is the only diverging test — although I don't think it should block this PR from being merged, as it's a behavior that can be currently reproduced in `trunk`

Reply:
I agree.

>`// TODO: investigate why `onChange` gets called twice instead of once`

I guess it's been that way for quite some time now and somehow the async oversights in the test suite were somehow masking it. We could look at extracting [this change](https://github.com/WordPress/gutenberg/pull/40568/commits/e662bdef95ffb7318cb899890e1252022e0ed266#diff-b21e5b36bcbb96cd162206795bc262d1ecb0b2ccb1bac9f0096d13c4a1c9c03a)(and your added code comment) from #40568 to a separate PR.

Reply:
Sounds like a plan! Opened #40796 with those changes